packages/flutter_tools/test/host_cross_arch.shard/macos_content_validation_test.dart

```diff
@@ -164,10 +181,9 @@ void main() {
         ),
       );
 
-      // Check read/write permissions are being correctly set
-      final String rawStatString = outputFlutterFramework.statSync().modeString();
```

Comment:
`modeString` docs say:

> Extra permission bits may be represented by prepending "(suid)", "(guid)", and/or "(sticky)" to the mode string.

Instead of doing string parsing, use `mode` int and convert to octal (755).

Reply:
i wonder why the engine artifact permissions 40755 but the framework permissions are 4755?

Reply:
Whoops, typo, they should all be `40755`.  These don't pass until https://github.com/flutter/engine/pull/52961 rolls into the framework, at which point I'll rebase this on top and hopefully the tests will pass.